File: packages/ui-compose/ui/src/main/kotlin/com/kinetixui/ui/Avatar.kt
package com.kinetixui.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.res.dimensionResource
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * KinetixAvatar / KinetixAvatarFallback — mirror
 * `packages/ui/src/components/avatar.tsx`'s `Avatar` / `AvatarFallback`
 * (`size-10 rounded-full` container, `bg-muted text-sm` fallback). 40dp is a
 * fixed Figma size, not on the shared `spacing_*` scale — hardcoded.
 *
 * `AvatarImage` isn't ported: it's a bare `<img>` wrapper in the React
 * source with no design-specific styling beyond `aspect-square size-full`,
 * so it's just `Image(..., modifier = Modifier.fillMaxSize())` inside
 * [KinetixAvatar]'s `content` slot — no reason to wrap it.
 *
 * `AvatarGroup` (stacked, overflow "+N" marker) also isn't ported: it needs
 * to inspect and re-wrap its children, which isn't idiomatic Compose the
 * way `React.Children.toArray` is in React. Compose it directly instead:
 * `Row(horizontalArrangement = Arrangement.spacedBy((-8).dp)) { KinetixAvatar { ... } }`.
 */
@Composable
fun KinetixAvatar(
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    Box(
        modifier = modifier.size(40.dp).clip(CircleShape),
        contentAlignment = Alignment.Center,
    ) {
        content()
    }
}

@Composable
fun KinetixAvatarFallback(
    text: String,
    modifier: Modifier = Modifier,
) {
    val colors = KinetixColorScheme.current
    Box(
        modifier = modifier.fillMaxSize().background(colors.muted),
        contentAlignment = Alignment.Center,
    ) {
        Text(
            text = text,
            color = colors.mutedForeground,
            fontSize = dimensionResource(R.dimen.font_size_body_md).value.sp,
        )
    }
}

File: packages/ui-swiftui/Sources/KinetixUI/Avatar.swift
//
// Avatar.swift — KinetixAvatar / KinetixAvatarFallback.
//
// Mirror packages/ui/src/components/avatar.tsx's `Avatar` /
// `AvatarFallback` (`size-10 rounded-full` container, `bg-muted text-sm`
// fallback). 40 is a fixed Figma size, off the shared spacing scale.
//
// `AvatarImage` isn't ported — it's a bare `<img>` with no design-
// specific styling; pass `Image(…).resizable().scaledToFill()` into the
// `content` slot. `AvatarGroup` isn't ported either (it re-wraps its
// children, not idiomatic in SwiftUI) — compose it directly:
// `HStack(spacing: -8) { KinetixAvatar { … } }`, same call as the
// Compose port.
//

import SwiftUI

public struct KinetixAvatar<Content: View>: View {
    private let size: CGFloat
    private let content: Content

    public init(size: CGFloat = 40, @ViewBuilder content: () -> Content) {
        self.size = size
        self.content = content()
    }

    public var body: some View {
        content
            .frame(width: size, height: size)
            .clipShape(Circle())
    }
}

public struct KinetixAvatarFallback: View {
    @Environment(\.kinetixColors) private var colors
    private let text: String

    public init(_ text: String) {
        self.text = text
    }

    public var body: some View {
        colors.muted
            .overlay {
                Text(text)
                    .font(.kinetixBody)
                    .foregroundStyle(colors.mutedForeground)
            }
    }
}
